src/Microsoft.AspNetCore.SignalR.Common/Internal/Formatters/BinaryMessageFormatter.cs

```diff
@@ -31,5 +31,44 @@ public static void WriteLengthPrefix(long length, IBufferWriter<byte> output)
 
             output.Write(lenBuffer.Slice(0, lenNumBytes));
         }
+
+        public static int WriteLengthPrefix(long length, Span<byte> output)
+        {
+            // This code writes length prefix of the message as a VarInt. Read the comment in
+            // the BinaryMessageParser.TryParseMessage for details.
+
+            Span<byte> lenBuffer = stackalloc byte[5];
```

Comment:
Why use a stack buffer and copy rather than the buffer we were provided? It has to be at least 5 bytes long to accept the longest varint.

Reply:
Done